low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -314,11 +326,13 @@ def validate(self):
             raise SetupError(1, msg)
         logger.info("%s - Validation done" % self.name)
 
-        if self.args.coverage and 'self' in self.args.plan[0]:
-            raise SetupError(errno.EINVAL,
-                             "%s - Argument validation failure. %s",
-                             self.name,
-                             "Code coverage can not be enabled with self tests.")
+        if self.args.coverage and (self.args.plan[0] in IVT_TEST_PLANS or
+            'self' in self.args.plan[0]):
```

Comment:
Only sanity (new) needs to be avoided right for coverage, why others being restricted here? Also self is no more a valid plan, it should be discarded if passed still

@sumedhak27 can you double check and add your suggestions

Reply:
keeping the check only for TEST_REQ_SERVICE_RESTART list, it contains test plans for which we restart service.